### PR TITLE
Support loading additional certificates

### PR DESCRIPTION
**Pull request by mjg59**
Based on an idea by @vathpela, support loading certificates from additional EFI binaries as long as those binaries are themselves signed by an already trusted certificate.

### CODE REVIEW

File: src/shim.c [hunk 1]
```diff
@@ -1825,10 +1875,10 @@ EFI_STATUS start_image(EFI_HANDLE image_handle, CHAR16 *ImagePath)
 			       efi_status);
 			return efi_status;
 		}
-		data = sourcebuffer;
-		datasize = sourcesize;
+		*data = sourcebuffer;
+		*datasize = sourcesize;
 #if  defined(ENABLE_HTTPBOOT)
-	} else if (find_httpboot(li->DeviceHandle)) {
+	} else if (find_httpboot(*li->DeviceHandle)) {
```
Comment (aburmash) [inline src/shim.c:1881]: } else if (find_httpboot((*li)->DeviceHandle)) {
 brackets missing ?

Comment (mjg59) [reply src/shim.c:1881]: Oh gosh, good catch.

File: src/shim.c [hunk 2]
```diff
@@ -1880,7 +1895,7 @@ EFI_STATUS start_image(EFI_HANDLE image_handle, CHAR16 *ImagePath)
 	 */
 	efi_status = handle_image(data, datasize, li, &entry_point,
 				  &alloc_address, &alloc_pages);
-	if (EFI_ERROR(efi_status)) {
+	if (efi_status != EFI_SUCCESS) {
```
Comment (lcp) [inline src/shim.c:1954]: This change looks not necessary.

Comment (mjg59) [reply src/shim.c:1954]: It's explained in the commit message - without this change, build fails with:

.././src/shim.c: In function ‘start_image’:
.././src/shim.c:1966:15: error: ‘entry_point’ may be used uninitialized in this function [-Werror=maybe-uninitialized]

Comment (lcp) [reply src/shim.c:1954]: Ah, thanks. Didn't realize this it related to the error mentioned in the commit message.

Comment (mjg59) [reply src/shim.c:1954]: I'll rewrite the commit message to clarify that, thank you!

**Comment by vathpela** [comment]
This was auto-closed because I can't re-assign it to rhboot:main and for some reason it didn't follow when the default branch was renamed. @mjg59 can you please edit this to make it be against rhboot:main and re-open it?

**Comment by mjg59** [comment]
I've moved on from Google and am not actively working on this at the moment, but for anyone else working on it: Google retains copyright to this code, but I was authorised to release it under the license terms covering the rest of Shim.

**Pull request closed without merging**